src/behaviours/tray/clicking-tray-menu-item-originating-from-extension.test.ts

```diff
@@ -0,0 +1,134 @@
+  let applicationBuilder: ApplicationBuilder;
+  let logErrorMock: jest.Mock;
+
+  beforeEach(async () => {
+    applicationBuilder = getApplicationBuilder();
+
+    applicationBuilder.beforeApplicationStart(({ mainDi }) => {
+      logErrorMock = jest.fn();
+
+      mainDi.override(loggerInjectable, () => ({ error: logErrorMock }) as unknown as Logger);
+    });
+
+    await applicationBuilder.render();
+  });
+
+  describe("when extension is enabled", () => {
+    let someExtension: SomeTestExtension;
+    let clickMock: jest.Mock;
+
+    beforeEach(async () => {
+      clickMock = jest.fn();
+
+      someExtension = new SomeTestExtension({
+        id: "some-extension-id",
+        trayMenus: [{ label: "some-label", click: clickMock }],
+      });
+
+      await applicationBuilder.extensions.main.enable(someExtension);
+    });
+
+    it("when item is clicked, triggers the click handler", () => {
+      applicationBuilder.tray.click(
+        "some-label-tray-menu-item-for-extension-some-extension-id-instance-1",
+      );
+
+      expect(clickMock).toHaveBeenCalled();
+    });
+
+    describe("given click handler throws synchronously, when item is clicked", () => {
+      beforeEach(() => {
+        clickMock.mockImplementation(() => {
+          throw new Error("some-error");
+        });
+
+        applicationBuilder.tray.click(
+          "some-label-tray-menu-item-for-extension-some-extension-id-instance-1",
+        );
+      });
+
+      it("logs the error", () => {
+        expect(logErrorMock).toHaveBeenCalledWith(
+          '[TRAY]: Clicking of tray item "some-label" from extension "some-extension-id" failed.',
+          expect.any(Error),
+        );
+      });
+    });
+
+    describe("given click handler rejects asynchronously, when item is clicked", () => {
+      beforeEach(() => {
+        clickMock.mockImplementation(() => Promise.reject("some-rejection"));
+
+        applicationBuilder.tray.click(
+          "some-label-tray-menu-item-for-extension-some-extension-id-instance-1",
+        );
+      });
+
+      it("logs the error", () => {
+        expect(logErrorMock).toHaveBeenCalledWith(
+          '[TRAY]: Clicking of tray item "some-label" from extension "some-extension-id" failed.',
+          "some-rejection",
+        );
+      });
+    });
+
+    describe("when disabling extension", () => {
+      beforeEach(() => {
+        applicationBuilder.extensions.main.disable(someExtension);
+      });
+
+      it("does not have the tray menu item from extension", () => {
+        applicationBuilder.extensions.main.disable(someExtension);
+
+        expect(
+          applicationBuilder.tray.get(
+            "some-label-tray-menu-item-for-extension-some-extension-id-instance-1",
+          ),
+        ).toBeNull();
+      });
+
+      // Note: Motivation here is to make sure that enabling same extension does not throw
+      it("when extension is re-enabled, has the tray menu item from extension", async () => {
```

Comment:
And then shouldn't this also be in another `describe`

Reply:
No need if there is only one test. We could add second test just for documentative purpose. But it would be empty test, so comment does same trick.